### ISSUE TITLE
reward score is too high

### ISSUE THREAD

**Issue by Aida-yy** [huggingface/trl#691]
I use rewardTrainer train my reward model in SHP dataset with 1 epoch, but reward score is very large, most are 60-70, other reward model outputs are centered around zero, with both positive and negative values. 
Why is this the case? How can I resolve this?

**Comment by younesbelkada** [comment]
Hi @Aida-yy 
Thanks for the issue, on my side I am not sure how it is the case, you can normalize the scores to be centered around a certain value but from a previous offline discussion with @lvwerra I think this does not really matter in practice for PPO but I am not sure; cc @lvwerra @vwxyzjn

**Comment by Aida-yy** [comment]
Thanks. maybe is not really matter in practice for PPO, however, I'm just curious why this would lead to such a situation. It seems other reward models don't incorporate normalization during training either. Also, my reward model gives almost identical scores to similar answers, so the reward model lacks discriminativity.

**Comment by younesbelkada** [comment]
Ok I see, can you share how exactly do you retrieve your reward score in your case?

**Comment by Aida-yy** [comment]
@younesbelkada Sure.  here is some code:
```
# modeling
outputs = self.model(
  input_ids=input_ids,
  attention_mask=attention_mask,
  position_ids=position_ids,
  past_key_values=past_key_values,
  inputs_embeds=inputs_embeds,
  use_cache=use_cache,
  output_attentions=output_attentions,
  output_hidden_states=output_hidden_states,
  return_dict=return_dict,
)
hidden_states = outputs[0]  # size = (B, L, E)
scores = self.score_head(hidden_states)  # size = (B, L, D)

end_scores = []
for i in range(input_ids.size(0)):
    end_index = attention_mask[i].nonzero()[-1].item()
    end_scores.append(scores[i, end_index])  # size = (D,)
end_scores = torch.stack(end_scores, dim=0)  # size = (B, D)

if not return_dict:
    return scores, end_scores

return ScoreModelOutput(
    scores=scores,  # size = (B, L, D)
    end_scores=end_scores,  # size = (B, D)
)
###########################################
# get reward score
batch = tokenizer(text, return_tensors="pt", padding=True, truncation=True, max_length=2048)

with torch.no_grad():
    reward = reward_model(
        batch["input_ids"].cuda(), attention_mask=batch["attention_mask"].cuda()
    ).end_scores.squeeze(-1)
    reward_score = reward.detach().cpu().tolist()
```

**Comment by younesbelkada** [comment]
Thanks @Aida-yy , can you share how you have defined `reward_model` ?

**Comment by Aida-yy** [comment]
> Thanks @Aida-yy , can you share how you have defined `reward_model` ?

`reward_model = LlamaRewardModel.from_pretrained(model_name_or_path)`
where llamaRewardModel is # modeing

**Comment by Aida-yy** [comment]
The latest observation is that this problem is caused by the training process. As my training steps keep increasing, the reward also keeps increasing. Until the loss stops changing.
